spec/services/courses/add_students_service_spec.rb

```diff
@@ -127,5 +127,30 @@
         expect(current_email).to eq(nil)
       end
     end
+
+    context 'notification is enabled' do
+      let(:notify) { true }
+      let(:school) { course.school }
+
+      it 'send email to user set his first password' do
+        service = double('Users::MailFirstPasswordTokenService')
+        allow(Users::MailFirstPasswordTokenService).to receive(:new) { service }
+        expect(service).to receive(:execute)
+
+        subject.add([student_1_data])
```

Comment:
This test just test if the service double is executed. This is not really an integration test. Will not allow me for fearless refactoring of this feature :(

Reply:
Now i see the issue to test without dependency injection.
But as it's planned to remove this mail service, would it be more reassuring to just add an expectation to the service initialization?